python/lsst/ts/ess/csc/ess_csc.py

```diff
@@ -22,14 +22,16 @@
 __all__ = ["EssCsc"]
 
 import asyncio
-import json
 import math
-from types import SimpleNamespace
-from typing import Any, Dict, List, Optional, Union
+import traceback
+import types
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import jsonschema
 
 from .config_schema import CONFIG_SCHEMA
 from . import __version__
-from lsst.ts import salobj, tcpip, utils
+from lsst.ts import salobj, utils
 from lsst.ts.ess import common
 from lsst.ts.idl.enums.ESS import ErrorCode
```

Comment:
Here is a simple alternative to make the code backwards compatible with the current cycle:

```
try:
    from lsst.ts.idl.enums.ESS import ErrorCode
except ImportError:
    warning.warn("Running in ts-idl < 3.5 compatibility mode....")
    class ErrorCode(IntEnum):
        """Error codes that can be used by the ESS CSC."""


        NotConnected = 1
        AlreadyConnected = 2
        NotConfigured = 3
        BadConfiguration = 4
        ReadLoopFailed = 5
        TelemetryError = 6
        HardwareError = 7
```

Reply:
Of course, this is not going to work because what changed was the enum itself...

Reply:
But could be something on these lines..  You could even do something like:

```
# TODO: Use enum from ts-idl when version >3.7 is released.
class ErrorCode(IntEnum):
    """Error codes that can be used by the ESS CSC.

    Values:

    * ConnectionFailed: a model could not connect its data server.
      Perhaps the model configuration is wrong or the server is down.
    * ConnectionLost: a model lost its connection to its data server.
    * StartFailed: one or more models failed to start.
    * RunFailed: one or more models failed while running.
    """
    ConnectionFailed = 1
    ConnectionLost = 2
    StartFailed = 3
    RunFailed = 4
```

Anything to maintain backward compatibility with a newly released cycle would be welcomed!

Reply:
@tribeiro yes I did as you suggested (before seeing your suggestion): I added a local ErrorCode class to ts_ess_csc and filed a ticket to remove it when ts_idl 3.7 is available.